lesson07.md

```diff
@@ -0,0 +1,301 @@
+<!-- .slide: id="loops" -->
+
+# Loops
+
+---
+
+### The problem
+
+Output your name to `console.log` two times.
+
+```js
+console.log('Bob');
+console.log('Bob');
+```
+
+<!-- .element: class="fragment" -->
+
+Output your name to `console.log` 14207 times.
+
+<!-- .element: class="fragment" -->
+
+```js
+// huh ????
+```
+
+<!-- .element: class="fragment" -->
+
+---
+
+### Loops
+
+We use loops to run the same code in our programs repeatedly.
+
+More generally, a loop is a way to repeat an action multiple times as long as a certain condition is `true`.
+
+---
+
+### Loops: cake recipe
+
+Let's say I want to bake a cake:
+
+<div style="display: flex;">
+
+![scale](images/scale.webp) <!-- .element: style="width: 400px"-->
+
+<div>
+Cake recipe:<br>
+
+- 6 eggs
+- 3 cups of sugar
+- 5 cups of flour
+</div>
+</div>
+
+---
+
+### Loops: cake recipe
+
+We take an empty bowl and keep our ingredients ready.
+
+We need 5 cups of flour.
+
+1. What's the action that we need to repeat?
+1. For how long do we need to repeat it?
+
+---
+
+### Loops: cake recipe
+
+1. The action we need to repeat is adding a cup of flour to the bowl
+1. We repeat it 5 times
+
+---
+
+### Loops: cake recipe
+
+We can distinguish different parts in a loop:
+
+- initialization: a starting point
+- condition: a test to see if the loop should stop
+- update: changes the value checked by the condition
+- action: what we want to repeat
+
+---
+
+### Loops: cake recipe
+
+- initialization: no flour in the bowl (`cupsOfFlour = 0`)
+- condition: did we put less than 5 cups of flour? (`cupsOfFlour < 5`)
+- update: cups of flour in the bowl (`cupsOfFlour++`)
```

Comment:
I think we didn't introduce ++ and -- yet. You could explain it during the course or add last semesters slides

Reply:
I will explain it today, I thought it's a little thing to add for today's content as I will only cover while-loop